src/toil_scripts/tools/preprocessing.py

```diff
@@ -233,33 +388,44 @@ def run_base_recalibration(job, cores, indel_bam, indel_bai, ref, ref_dict, fai,
     :param str fai: Reference index FileStoreID
     :param str dbsnp: DBSNP VCF FileStoreID
     :param str mem: Memory value to be passed to children. Needed for CI tests
+    :param str bqsr: BQSR recalibration table FileStoreID
     :param bool unsafe: If True, runs gatk UNSAFE mode: "-U ALLOW_SEQ_DICT_INCOMPATIBILITY"
     :return: FileStoreID for the processed bam
     :rtype: str
     """
     work_dir = job.fileStore.getLocalTempDir()
-    file_ids = [ref, fai, ref_dict, indel_bam, indel_bai, dbsnp]
-    inputs = ['ref.fasta', 'ref.fasta.fai', 'ref.dict', 'sample.indel.bam', 'sample.indel.bai', 'dbsnp.vcf']
+    file_ids = [ref, fai, ref_dict, indel_bam, indel_bai, dbsnp, mills]
+    inputs = ['ref.fasta', 'ref.fasta.fai', 'ref.dict', 'sample.bam', 'sample.bai',
+              'dbsnp.vcf', 'mills.vcf']
     for file_store_id, name in zip(file_ids, inputs):
         job.fileStore.readGlobalFile(file_store_id, os.path.join(work_dir, name))
     # Call: GATK -- IndelRealigner
     parameters = ['-T', 'BaseRecalibrator',
                   '-nct', str(cores),
                   '-R', '/data/ref.fasta',
-                  '-I', '/data/sample.indel.bam',
+                  '-I', '/data/sample.bam',
                   '-knownSites', '/data/dbsnp.vcf',
-                  '-o', '/data/sample.recal.table']
+                  '-knownSites', '/data/mills.vcf',
+                  '-o', '/data/recal_data.table']
+    if bqsr:
+        # Best practices recommends doing two passes
+        # https://software.broadinstitute.org/gatk/documentation/article?id=2801
+        job.fileStore.readGlobalFile(bqsr, os.path.join(work_dir, 'initial_recal_data.table'))
+        parameters += ['-BQSR', '/data/initial_recal_data.table']
+
     if unsafe:
         parameters.extend(['-U', 'ALLOW_SEQ_DICT_INCOMPATIBILITY'])
     docker_call(tool='quay.io/ucsc_cgl/gatk:3.5--dba6dae49156168a909c43330350c6161dc7ecc2',
-                inputs=inputs,
-                outputs={'sample.recal.table': None},
-                work_dir=work_dir, parameters=parameters, env=dict(JAVA_OPTS='-Xmx{}'.format(mem)))
+                inputs=inputs, mock=mock,
+                outputs={'recal_data.table': None},
+                work_dir=work_dir, parameters=parameters,
+                env=dict(_JAVA_OPTIONS='-Djava.io.tmpdir=/data/ -Xmx{}'.format(mem)))
     # Write output to file store
-    return job.fileStore.writeGlobalFile(os.path.join(work_dir, 'sample.recal.table'))
+    return job.fileStore.writeGlobalFile(os.path.join(work_dir, 'recal_data.table'))
 
 
-def run_print_reads(job, cores, table, indel_bam, indel_bai, ref, ref_dict, fai, mem, unsafe=False):
+def run_print_reads(job, cores, table, indel_bam, indel_bai, ref, ref_dict, fai, mem,
```

Comment:
Also, while this does run `PrintReads`, I think it is more accurate to call this function `recalibrate_reads`, since you can run `PrintReads` in other invocations (e.g., for downsampling).